Lib/pathlib.py

```diff
@@ -609,13 +576,20 @@ def parts(self):
         else:
             return tuple(self._tail)
 
+    def makepath(self, *args):
+        """Construct a new path object from any number of path-like objects.
+        Subclasses may override this method to customize how new path objects
+        are created from methods like `iterdir()`.
+        """
+        return type(self)(*args)
```

Comment:
Hmm, wouldn't this be simpler as a classmethod?

```suggestion
    @classmethod
    def makepath(cls, *args):
        """Construct a new path object from any number of path-like objects.
        Subclasses may override this method to customize how new path objects
        are created from methods like `iterdir()`.
        """
        return cls(*args)
```

Reply:
Making it a classmethod means that you can't share state between paths no?

Reply:
I see -- sorry, just getting back to pathlib stuff after a while away!

I guess I find the "makepath" name quite unintuitive in that case. The name to me implies that it's "just" an alternative constructor, which I'd expect to be a classmethod. Maybe something like "sproutpath" (feel free to bikeshed the name), which is clearer that it is _using the state of the current `Path` instance_ would be better?

Reply:
`with_segments()`, perhaps? Complements `with_name()` etc.

Reply:
`with_segments` sounds like it's kiiinda "the previous path's segements, but with some new ones added at the end".

What about "newpath"? I think that reinforces the fact that the returned object has completely different segments to the old one.

Reply:
Although this method is mostly called under-the-hood, it may also be useful when dealing with subclasses of pathlib classes. For example:

```python
from tarfile import TarFile, TarPath
readme_path = TarPath('README.txt', tarfile=TarFile('blah.tar.gz'))
license_path = readme_path.newpath('LICENSE.txt')
```

I mention this just in case it helps us with the naming question. In the above example, `readme_path.newpath()` is a sort of factory for path objects sharing the same underlying `TarFile` object. And I suppose factories _make_ things. But I don't mind `newpath()` :)

Reply:
Of the options floated so far, I like the "newpath" name the best. But it's your module now, afterall, so I don't feel like I should have the final call here :D

You could maybe start a thread on Discord or Discuss if you'd like the opinion of more core devs?

Reply:
Thread created here! https://discuss.python.org/t/bikeshedding-opportunity-help-name-a-pathlib-method/26186